diff --git a/typings/types.d.cts b/typings/types.d.cts
index 03f61a5a03afbbfd93ea958ab3fc984c745c117b..9e2848eb2ef806e8e8579ba61db3e0e55183119b 100644
--- a/typings/types.d.cts
+++ b/typings/types.d.cts
@@ -101,7 +101,9 @@ export type Circular<TJSONSchema extends JSONSchema> = TJSONSchema extends {
     properties: {
         [key: string]: JSONSchema;
     };
-} ? TJSONSchema extends PropertyValue<TJSONSchema, Property<TJSONSchema>> ? true : Circular<PropertyValue<TJSONSchema, Property<TJSONSchema>>> : false;
+} ? TJSONSchema extends PropertyValue<TJSONSchema, Property<TJSONSchema>> ? true : Circular<PropertyValue<TJSONSchema, Property<TJSONSchema>>> : TJSONSchema extends { items: JSONSchema }
+? true
+: false;
 export type Property<TJSONSchema extends JSONSchema> = keyof TJSONSchema['properties'];
 export type PropertyValue<TJSONSchema extends JSONSchema, TProperty extends keyof TJSONSchema['properties']> = TJSONSchema['properties'][TProperty];
 export type FromSchema<T> = T extends {
diff --git a/typings/types.d.ts b/typings/types.d.ts
index d5b70bb84250dac31195e12fd4f3dee922e16043..d76f827a7b34ee448ef2cb8427efc4de81c240cb 100644
--- a/typings/types.d.ts
+++ b/typings/types.d.ts
@@ -101,7 +101,9 @@ export type Circular<TJSONSchema extends JSONSchema> = TJSONSchema extends {
     properties: {
         [key: string]: JSONSchema;
     };
-} ? TJSONSchema extends PropertyValue<TJSONSchema, Property<TJSONSchema>> ? true : Circular<PropertyValue<TJSONSchema, Property<TJSONSchema>>> : false;
+} ? TJSONSchema extends PropertyValue<TJSONSchema, Property<TJSONSchema>> ? true : Circular<PropertyValue<TJSONSchema, Property<TJSONSchema>>>   : TJSONSchema extends { items: JSONSchema }
+? true
+: false;
 export type Property<TJSONSchema extends JSONSchema> = keyof TJSONSchema['properties'];
 export type PropertyValue<TJSONSchema extends JSONSchema, TProperty extends keyof TJSONSchema['properties']> = TJSONSchema['properties'][TProperty];
 export type FromSchema<T> = T extends {
